packages/host/app/components/matrix/forgot-password.gts

```diff
@@ -0,0 +1,447 @@
+      email: this.email,
+      sendAttempt: 1,
+    };
+    this.sendEmailValidationTask.perform();
+  }
+
+  @action
+  private resendEmailValidation() {
+    if (this.state.type !== 'waitForEmailValidation') {
+      throw new Error(
+        `invalid state: cannot resendEmailValidation() in state ${this.state.type}`,
+      );
+    }
+
+    this.state = {
+      ...this.state,
+      sendAttempt: this.state.sendAttempt++,
+    };
+    this.sendEmailValidationTask.perform();
+  }
+
+  private sendEmailValidationTask = restartableTask(async () => {
+    if (
+      this.state.type === 'initial' ||
+      this.state.type === 'resetPassword' ||
+      this.state.type === 'resetPasswordSuccess'
+    ) {
+      throw new Error(
+        `invalid state: cannot sendEmailValidation() in state ${this.state.type}`,
+      );
+    }
+
+    try {
+      let clientSecret = uuidv4();
+      let { sid } = await this.matrixService.client.requestPasswordEmailToken(
+        this.state.email,
+        clientSecret,
+        this.state.sendAttempt,
+      );
+      this.state = {
+        ...this.state,
+        type: 'waitForEmailValidation',
+        clientSecret,
+        sid,
+      };
+    } catch (e: any) {
+      if (isMatrixError(e)) {
+        this.emailError = 'No account with the given email address exists';
+      }
+      if (this.state.type === 'validateEmail') {
+        this.state = {
+          type: 'initial',
+        };
+      }
+
+      throw e;
+    }
+  });
+
+  @action
+  private continueToResetPassword() {
+    if (this.state.type !== 'waitForEmailValidation') {
+      throw new Error(
+        `invalid state: cannot continueToResetPassword() in state ${this.state.type}`,
+      );
+    }
+
+    this.state = {
+      ...this.state,
+      type: 'resetPassword',
+    };
+  }
+
+  private resetPassword = restartableTask(async () => {
+    if (this.state.type !== 'resetPassword') {
+      throw new Error(
+        `invalid state: cannot resetPassword() in state ${this.state.type}`,
+      );
+    } else if (!this.password) {
+      throw new Error(
+        `bug: should never get here: reset password button disabled when no password`,
+      );
+    }
+
+    try {
+      await this.matrixService.client.setPassword(
+        {
+          threepid_creds: {
+            sid: this.state.sid,
+            client_secret: this.state.clientSecret,
+          },
+          type: 'm.login.email.identity',
+        },
+        this.password,
+      );
+      this.state = {
+        ...this.state,
+        type: 'resetPasswordSuccess',
+      };
+    } catch (e: any) {
+      if (isMatrixError(e)) {
+        this.error = 'Please check your email to validate reset password';
+        setTimeout(() => (this.error = undefined), 2000);
+      }
+
+      throw e;
+    }
+  });
+}
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface ForgotPassword {
+    'Matrix::ForgotPassword': typeof ForgotPassword;
+  }
+}
```

Comment:
i think we don't need to do this any more since we are using routeable templates--is that right @backspace?

Reply:
agreed 🎉